server/src/main/java/org/candlepin/controller/EntitlementCertificateGenerator.java

```diff
@@ -390,9 +389,16 @@ public void regenerateCertificatesOf(Collection<Owner> owners, Collection<Produc
         // TODO: This is a very expensive operation. Update pool curator with something to let us
         // do this without hitting the DB several times over.
         for (Owner owner : owners) {
-            pools.addAll(
-                this.poolCurator.listAvailableEntitlementPools(null, owner, productIds, now)
-            );
+            if (lazy) {
+                for (Product product : products) {
+                    poolCurator.markCertificatesDirtyForOwnerProductId(owner, product.getId());
```

Comment:
We could optimize this a bit by modifying this to build up our set of product IDs, then issue one update statement (and disregarding my request to drop the IN on the inner query below).